src/main/java/com/sonyericsson/jenkins/plugins/bfa/providers/FailureCauseProvider.java

```diff
@@ -46,19 +47,22 @@ public void provideCompletedRunData(Run run, JSONObject json) {
         FailureCauseBuildAction action = run.getAction(FailureCauseBuildAction.class);
         if (action != null) {
             List<FoundFailureCause> foundFailureCauses = action.getFoundFailureCauses();
-            JSONObject failureCausesJSON = new JSONObject();
+            JSONArray failureCausesJSON = new JSONArray();
             for (FoundFailureCause foundFailureCause : foundFailureCauses) {
                 JSONObject failureCauseJSON = new JSONObject();
+                failureCauseJSON.put("id", foundFailureCause.getId());
                 failureCauseJSON.put("name", foundFailureCause.getName());
                 failureCauseJSON.put("description", foundFailureCause.getDescription());
                 failureCauseJSON.put("categories", foundFailureCause.getCategories());
-                JSONObject foundIndicationsJSON = new JSONObject();
+                JSONArray foundIndicationsJSON = new JSONArray();
                 for (FoundIndication ind : foundFailureCause.getIndications()) {
-                    foundIndicationsJSON.put("pattern", ind.getPattern());
-                    foundIndicationsJSON.put("matchingString", ind.getMatchingString());
+                    JSONObject foundIndicationJSON = new JSONObject();
```

Comment:
_nit_ With just the difference of an `s` in the variable name it makes it a bit hard to read.